Faq Pattern Section

esrat71 opened the pull request:
Faq section designed

**Description**

faq section designed as pattern

Reference to the issue that the PR fixes:
https://github.com/WordPress/twentytwentyfour/issues/119

**Screenshots**
https://prnt.sc/CDb7GJTljavD

**Testing Instructions**

1.  Activate the theme.
2.  Create a pettern named faq
3.  Add the code viewing as code editor.
4.  See the changes

**Another way**

1.     copy the code
2.     Create a faq php file and see the changes

esrat71 committed:
Faq Pattern Section

Faq section designed

carolinan commented:
Thank you. A few things:

In the pull request description please add a reference to the issue that the PR fixes:
https://github.com/WordPress/twentytwentyfour/issues/119

All texts need to use translation functions. Please see the documentation here: https://developer.wordpress.org/apis/internationalization/

The pattern needs to use a list block with a custom style, not paragraphs and HRs.

carolinan commented:
Here is an example of how to add a custom style for the list block:
https://github.com/WordPress/twentytwentyfour/pull/150/files#diff-01c9525afc2c87cfcb03124117c9d0ebb067029f6c955a607fa0fd4274aca556R70

esrat71 commented:
Okay, Thank you for review. I am going to make it with list item with custom style. Here's a query for custom style, I f i add a register block style , may need to add css, then should i do that in style css file or what's the procedure actually.

carolinan commented:
Add it to style.css for now.

huzaifaalmesbah commented on patterns/faq.php:
```diff
@@ -0,0 +1,44 @@
+
+<!-- wp:group {"style":{"color":{"background":"#101010"},"spacing":{"padding":{"top":"8vh","bottom":"8vh","left":"6px","right":"6px"}}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group has-background" style="background-color:#101010;padding-top:8vh;padding-right:6px;padding-bottom:8vh;padding-left:6px"><!-- wp:heading {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base-2"}}},"typography":{"fontSize":"10rem","letterSpacing":"-0.02em"}},"textColor":"base-2"} -->
+<h2 class="wp-block-heading has-base-2-color has-text-color has-link-color" style="font-size:10rem;letter-spacing:-0.02em">FAQs</h2>
```
Ready For Translation

huzaifaalmesbah commented on patterns/faq.php:
```diff
@@ -0,0 +1,44 @@
+<p class="has-base-2-color has-text-color has-link-color has-medium-font-size">↓ Can I apply to be a part of the team or work as a contractor?</p>
+<!-- /wp:paragraph --></div>
+<!-- /wp:group --></div>
+<!-- /wp:group -->
```
file must end with a newline

huzaifaalmesbah commented on patterns/faq.php:
```diff
@@ -0,0 +1,44 @@
+<!-- /wp:heading -->
+
+<!-- wp:group {"style":{"border":{"top":{"color":"#ffffff33","width":"1px"},"right":{"width":"0px","style":"none"},"bottom":{"color":"#ffffff33","width":"1px"},"left":{"width":"0px","style":"none"}},"spacing":{"padding":{"top":"15px","bottom":"15px"}}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group" style="border-top-color:#ffffff33;border-top-width:1px;border-right-style:none;border-right-width:0px;border-bottom-color:#ffffff33;border-bottom-width:1px;border-left-style:none;border-left-width:0px;padding-top:15px;padding-bottom:15px"><!-- wp:paragraph {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base-2"}}},"layout":{"selfStretch":"fit","flexSize ...
```
use contrast-3 for all border color

huzaifaalmesbah commented on patterns/faq.php:
```diff
@@ -0,0 +1,44 @@
+
+?>
+
+<!-- wp:group {"style":{"color":{"background":"#101010"},"spacing":{"padding":{"top":"8vh","bottom":"8vh","left":"6px","right":"6px"}}},"layout":{"type":"constrained"}} -->
```
use align: full width and  background-color: contrast

huzaifaalmesbah commented on patterns/faq.php:
```diff
@@ -0,0 +1,44 @@
+
+<!-- wp:group {"style":{"border":{"top":{"color":"#ffffff33","width":"1px"},"right":{"width":"0px","style":"none"},"bottom":{"color":"#ffffff33","width":"1px"},"left":{"width":"0px","style":"none"}},"spacing":{"padding":{"top":"15px","bottom":"15px"}}},"layout":{"type":"constrained"}} -->
+<div class="wp-block-group" style="border-top-color:#ffffff33;border-top-width:1px;border-right-style:none;border-right-width:0px;border-bottom-color:#ffffff33;border-bottom-width:1px;border-left-style:none;border-left-width:0px;padding-top:15px;padding-bottom:15px"><!-- wp:paragraph {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base-2"}}},"layout":{"selfStretch":"fit","flexSize ...
+<p class="has-base-2-color has-text-color has-link-color has-medium-font-size">↓ What is your process working in smaller projects?</p>
```
Prepare all content for translation

huzaifaalmesbah commented:
also use already registered pattern categories or register a new category faq.
<img width="338" alt="Screenshot 2023-08-31 at 10 53 49 AM" src="https://github.com/WordPress/twentytwentyfour/assets/63150399/f8f1d78f-9431-4cfe-b5bc-0f7096023661">

esrat71 commented:
@huzaifaalmesbah Thank you for reviewing. I am going to do the needful changes and will removed the category also. Will submit asap.

esrat71 committed:
added custom block style

Added custom block style in function , custom css and the main faq file with list block

esrat71 committed:
removed extra semicolon

esrat71 committed:
space indention removed

esrat71 committed:
tab indented

esrat71 committed:
added Translation and custom list style changed

Have translable all the hard coded string text and also changed the css file for list item

esrat71 committed:
Have removed indention

esrat71 commented:
hi , @carolinan , sorry to bother you . Can you please guide me. My PHP code is getting failed all time for indention issue. Kinda lost in somewhere. I have used tab as indention still getting error.

esrat71 committed:
have checked again

hope indention work fine

esrat71 committed:
update function code

esrat71 committed:
again tried for tab idention

juanfra commented:
Thanks for the contribution @esrat71!

One way to check what's going wrong with the code quality issues, is by checking the results of the [failed test for the PR](https://github.com/WordPress/twentytwentyfour/actions/runs/6037555800/job/16382059011?pr=158). You can also click on the [Files Changed](https://github.com/WordPress/twentytwentyfour/pull/158/files) tab for the PR to see the fails in their respective lines.

Another way could be running the test in your local environment. You can do that by running:

`composer install && composer run standards:check`

You can find more information about testing on the [project's readme](https://github.com/WordPress/twentytwentyfour#testing).

I also recommend you check how to use the translation functions in the context of patterns. You could probably check the conversation that [took place yesterday in Slack](https://wordpress.slack.com/archives/C02RP4VMP/p1693408967406689) to get more context on when to use `esc_html__` vs `esc_html_x`.

esrat71 commented:
> Thanks for the contribution @esrat71!
> 
> One way to check what's going wrong with the code quality issues, is by checking the results of the [failed test for the PR](https://github.com/WordPress/twentytwentyfour/actions/runs/6037555800/job/16382059011?pr=158). You can also click on the [Files Changed](https://github.com/WordPress/twentytwentyfour/pull/158/files) tab for the PR to see the fails in their respective lines.
> 
> Another way could be running the test in your local environment. You can do that by running:
> 
> `composer install && composer run standards:check`
> 
> You can find more information about testing on the [project's readme](https://github.com/WordPress/twentytwentyfour#testing).
> 
> I also recommend you check how to use the translation functions in the context of patterns. You could probably check the conversation that [took place yesterday in Slack](https://wordpress.slack.com/archives/C02RP4VMP/p1693408967406689) to get more context on when to use `esc_html__` vs `esc_html_x`.

Thank you. I am again going to check if i can resolve the issue

esrat71 committed:
some indention issue fixed

esrat71 committed:
remove paranthes spaces issue

carolinan commented on patterns/faq.php:
```diff
@@ -6,27 +6,28 @@
-    <div class="wp-block-group has-contrast-background-color has-background" style="padding-top:8vh;padding-right:6px;padding-bottom:8vh;padding-left:6px"><!-- wp:heading {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base-2"}}},"typography":{"fontSize":"10rem","letterSpacing":"-0.02em"}},"textColor":"base-2"} -->
-    <h2 class="wp-block-heading has-base-2-color has-text-color has-link-color" style="font-size:10rem;letter-spacing:-0.02em">FAQs</h2>
+<div class="wp-block-group has-contrast-background-color has-background" style="padding-top:8vh;padding-right:6px;padding-bottom:8vh;padding-left:6px"><!-- wp:heading {"style":{"elements":{"link":{"color":{"text":"var:preset|color|base-2"}}},"typography":{"fontSize":"10rem","letterSpacing":"-0.02em"}},"textColor":"base-2"} -->
+<h2 class="wp-block-heading has-base-2-color has-text-color has-link-color" style="font-size:10rem;letter-spacing:-0.02em"<?php echo esc_html__( 'FAQs?', 'twentytwentyfour' ); ?>></h2>
```
Please look carefully, this code is missing a closing tag and is broken.

esrat71 replied:
okay , checking again . Thank you

esrat71 replied:
i have checked after adding translable. strings its doesn't work properly. otherwise its work fine

esrat71 committed:
have fixed translable string issue in pattern.

have checked again , added all translable strings..

esrat71 committed:
have removed the space and added space

added and removed space after watching following brackets

esrat71 committed:
tab indentions checked again

MaggieCabrera commented on style.css:
```diff
@@ -33,3 +33,35 @@ a {
+.is-style-faq-check-lists li::before {
+    content: "\2193"; 
+    position: absolute;
+    left: 0;
```
Thank you so much for working on this. My main concern about this solution are RTL languages. In those cases we want the checkmark to be positioned to the right instead. [This](https://codex.wordpress.org/Right-to-Left_Language_Support) is the only official docs I found on RTL. It looks like we are going to need an extra stylesheet just for this. Do you know a better way @carolinan?

juanfra replied:
Another option could be using the [.rtl](https://github.com/WordPress/wordpress-develop/blob/6.3/src/wp-includes/post-template.php#L640) class that's automatically added to the body for RTL languages (if the idea is to avoid having a separate stylesheet).

carolinan replied:
I recently tried to use the .rtl class in a Gutenberg block and found that it is not used in the iframed block editor(!)
The editor uses `<html dir="rtl"> ` and we can not target children of elements with this attribute.
See https://developer.mozilla.org/en-US/docs/Web/CSS/:dir

carolinan replied:
Should we perhaps try to fix it in Gutenberg? https://github.com/WordPress/gutenberg/issues/51480

carolinan replied:
I am not sure we have enough time.

esrat71 replied:
Should i try to use in different way  without children ? like as just user can copy the icon and list item will as normal list item .

esrat71 replied:
Then User will have to icon like this and space between the item like the example https://prnt.sc/YnphTJPkKVnD What do you think ? as some of section in white background using this type pattern

esrat71 replied:
and one more thing, i tried to using marker and list style image, if i use list-style-position as normal , the list item border is not working as expected we want in design https://prnt.sc/avLaO4fbHqyR

MaggieCabrera committed:
changed spacings and swapped list block with the details block

MaggieCabrera commented:
I've made a few changes to this PR, the main one being that I changed the list block with the details block, so now we have the functionality of showing/hidding text for each FAQ item. I think this pattern is perfect to showcase this recently added block. Let's follow up with improvements and add some sample text to each of the FAQ items text. @beafialho can you provide some sample text for these, please?

MaggieCabrera committed:
add separator

beafialho commented:
Absolutely, I have also imagined this to use the details block. Here's some sample text, it can be used for all of the questions:

> Études offers comprehensive consulting, management, design, and research solutions. Our vision is to be at the forefront of architectural innovation, fostering a global community of architects and enthusiasts united by a passion for creating spaces. Every architectural endeavor is an opportunity to shape the future.
